httpcore5-h2/src/test/java/org/apache/hc/core5/http2/examples/Http2GreetingServer.java

```diff
@@ -0,0 +1,181 @@
+            return new BasicRequestConsumer<HttpEntity>(entityConsumer);
+
+        }
+
+        @Override
+        protected void handle(Message<HttpRequest, HttpEntity> requestMessage,
+                              AsyncServerRequestHandler.ResponseTrigger responseTrigger,
+                              HttpContext context) throws HttpException, IOException {
+
+            try {
+                final HttpCoreContext coreContext = HttpCoreContext.adapt(context);
+                final EndpointDetails endpoint = coreContext.getEndpointDetails();
+                final HttpRequest req = requestMessage.getHead();
+                final HttpEntity httpEntity = requestMessage.getBody();
+
+                // generic success response:
+                HttpResponse resp = new BasicHttpResponse(200);
+
+                // recording the request
+                System.out.println(String.format("[%s] %s %s %s", new Date(),
+                        endpoint.getRemoteAddress().toString(),
+                        req.getMethod(),
+                        req.getPath()));
+
+                // Request without an entity - GET/HEAD/DELETE
+                if (httpEntity == null) {
+                    responseTrigger.submitResponse(new BasicResponseProducer(resp), context);
+                    return;
+                }
+
+                // Request with an entity - POST/PUT
+                final ContentType entityContentType = ContentType.parse(httpEntity.getContentType());
+                String greeting = "Hello stranger\n";
+
+                if (entityContentType.getMimeType().equals(ContentType.APPLICATION_FORM_URLENCODED.getMimeType())) {
+
+                    Charset charset = ContentType.APPLICATION_FORM_URLENCODED.getCharset();
+                    if (entityContentType.getCharset() != null) {
+                        charset = entityContentType.getCharset();
+                    }
+
+                    StringBuilder contents = new StringBuilder();
+                    try (Reader reader = new BufferedReader(
+                            new InputStreamReader(httpEntity.getContent(), charset))) {
+                        int c;
+                        while ((c = reader.read()) != -1) {
+                            contents.append((char) c);
+                        }
+
+                    }
+
+                    // decoding the form entity into key/value pairs:
+                    List<NameValuePair> args = URLEncodedUtils.parse(contents.toString(),
+                            ContentType.APPLICATION_FORM_URLENCODED.getCharset());
+
+                    // composing greeting:
+                    greeting = String.format("Hello %s\n", args.get(0).getValue());
+                }
+
+                responseTrigger.submitResponse(new BasicResponseProducer(resp,
+                        new StringAsyncEntityProducer(greeting, ContentType.TEXT_PLAIN)), context);
+            } catch (Exception e) {
+                e.printStackTrace();
```

Comment:
@rferreira The exception handling code would also need to be reworked to something more presentable. Spitting exception stack traces to the console is not something we should be doing or recommending.